src/core/component.js

```diff
@@ -136,6 +126,48 @@ Component.prototype = {
     return utils.extend({}, data);
   },
 
+  updateCachedAttrValue: function (value) {
+    var isSinglePropSchema = isSingleProp(this.schema);
+    if (value === '') {
+      this.attrValue = undefined;
+      return;
+    }
+    if (typeof value === 'string') {
+      this.attrValue = this.parseAttrValueForCache(value);
+      return;
+    }
+    this.attrValue = value !== undefined ? extendProperties({}, value, isSinglePropSchema) : this.attrValue;
+  },
+
+  parseAttrValueForCache: function (value) {
```

Comment:
This could be a nicely scoped helper/private function

`function parseAttrValue (schema)`